src/components/PeopleFilters.tsx

```diff
@@ -1,21 +1,74 @@
-export const PeopleFilters = () => {
+import { useSearchParams } from 'react-router-dom';
+import React, { FC } from 'react';
+import classNames from 'classnames';
+import { SearchLink } from './SearchLink';
+
+export const PeopleFilters: FC = () => {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const sex = searchParams.get('sex') || '';
+  const query = searchParams.get('query') || '';
+  const centuries = searchParams.getAll('centuries') || [];
+
+  function updateSearch(params: { [key: string]: string | string[] | null }) {
+    Object.entries(params).forEach(([key, value]) => {
+      if (value === null) {
+        searchParams.delete(key);
+      } else if (Array.isArray(value)) {
+        searchParams.delete(key);
+
+        value.forEach(part => {
+          searchParams.append(key, part);
+        });
+      } else {
+        searchParams.set(key, value);
+      }
+    });
+
+    setSearchParams(searchParams);
+  }
+
+  const onQueryChange = (
+    { target: { value } }: React.ChangeEvent<HTMLInputElement>,
+  ) => {
+    updateSearch({ query: value.trim() || null });
+  };
+
   return (
     <nav className="panel">
       <p className="panel-heading">Filters</p>
 
       <p className="panel-tabs" data-cy="SexFilter">
-        <a className="is-active" href="#/people">All</a>
-        <a className="" href="#/people?sex=m">Male</a>
-        <a className="" href="#/people?sex=f">Female</a>
+        <SearchLink
+          params={{ sex: null }}
+          className={classNames({ 'is-active': sex === '' })}
+        >
+          All
+        </SearchLink>
+
+        <SearchLink
+          params={{ sex: 'm' }}
+          className={classNames({ 'is-active': sex === 'm' })}
+        >
+          Male
+        </SearchLink>
+
+        <SearchLink
+          params={{ sex: 'f' }}
+          className={classNames({ 'is-active': sex === 'f' })}
+        >
+          Female
+        </SearchLink>
```

Comment:
When you see a boilerplate code that duplicates itself then think about how to make it shorter. For example you can use object/enum where text is a key and the value is f/m/''. Then using map you can render any amount of links and have one object as config for it.